lib/theory/src/Theory/Constraint/System/JSON.hs

```diff
@@ -0,0 +1,394 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- |
+-- Copyright   : (c) 2016 Dominik Schoop
+-- License     : GPL v3 (see LICENSE)
+--
+-- Maintainer  : Dominik Schoop dominik.schoop@hs-esslingen.de
+-- Portability : GHC only
+--
+-- Conversion of the graph part of a sequent (constraint system) to a JSON graph format
+
+{-
+   ADVISE:
+   - DO NOT USE sequentToJSONPretty IN OPERATIONAL MODE.
+     Remember to replace sequentToJSONPretty with sequentToJSON in getTheoryGraphR in Handler.hs.
+   - The time consuming pretty printing of facts and terms can be disabeled by a parameter of
+     sequentToJSONPretty and sequentToJSON. Currently, everything is pretty with sequentToJSONPretty
+     and nothing is pretty with sequentToJSON.
+
+   TO DO:
+   - use console parameter to control whether DOT or JSON is used 
+   - generate JSON in non-interactive mode
+   - make it work for observational equivalence
+   - encode historic information in the graph: sequence of nodes added
+
+   OPEN PROBLEMS:
+   - JSON encodePretty converts < and > to "\u003c" and "\u003e"
+     ghastly postprocessing has been applied to sequentToJSONPretty
```

Comment:
Why is that a problem? These are the Unicode code-points for the LaTeX `\prec` symbols.